pkg/virt-handler/device-manager/socket_device.go

```diff
@@ -127,30 +89,39 @@ func (dpi *SocketDevicePlugin) Start(stop <-chan struct{}) (err error) {
 	}()
 
 	dpi.setInitialized(true)
-	logger.Infof("%s device plugin started", dpi.socketName)
+	logger.Infof("%s device plugin started", dpi.resourceName)
 	err = <-errChan
 
 	return err
 }
 
-// Stop stops the gRPC server
-func (dpi *SocketDevicePlugin) stopDevicePlugin() error {
-	defer func() {
-		if !IsChanClosed(dpi.done) {
-			close(dpi.done)
-		}
-	}()
+func NewSocketDevicePlugin(socketName, socketDir, socket string, maxDevices int) *SocketDevicePlugin {
+	dpi := &SocketDevicePlugin{
+		DevicePluginBase: &DevicePluginBase{
+			health:       make(chan deviceHealth),
+			resourceName: fmt.Sprintf("%s/%s", DeviceNamespace, socketName),
+			initialized:  false,
+			socketPath:   SocketPath(strings.Replace(socketName, "/", "-", -1)),
+			lock:         &sync.Mutex{},
+		},
 
-	// Give the device plugin one second to properly deregister
-	ticker := time.NewTicker(1 * time.Second)
-	defer ticker.Stop()
-	select {
-	case <-dpi.deregistered:
-	case <-ticker.C:
+		socket:     socket,
+		socketDir:  socketDir,
+		socketName: socketName,
 	}
-	dpi.server.Stop()
-	dpi.setInitialized(false)
-	return dpi.cleanup()
+
+	for i := 0; i < maxDevices; i++ {
+		deviceId := dpi.socketName + strconv.Itoa(i)
+		dpi.devs = append(dpi.devs, &pluginapi.Device{
+			ID:     deviceId,
+			Health: pluginapi.Healthy,
+		})
+	}
+	return dpi
+}
+
+func (dpi *SocketDevicePlugin) GetDeviceName() string {
+	return dpi.socketName
```

Comment:
leftover? Should be called from DevicePluginBase I think

Reply:
indeed, thanks for noticing. will remove it